### ISSUE TITLE
[FR] Reduce repetitive logs when patching metric names including an @-symbol (lightgbm)

### ISSUE THREAD

**Issue by itsmartinhi** [mlflow/mlflow#13463]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

When using the autolog feature with lightgbm (might also be the case for xgboost) while using ndcg@k as an evaluation metric, the metric name gets patched since "@" is not a valid symbol in metric names in mlflow.

Every time the metric name gets patched the actions gets logged with the "INFO" level.
This results in many repetitive logs since the patching action is part of the callback which gets called after each training iteration.

My proposal is to log this information once, instead of after every training iteration.

### Motivation

> #### What is the use case for this feature?
Non-repetitive logs

> #### Why is this use case valuable to support for MLflow users in general?
Less cluttered console output when training a model

> #### Why is this use case valuable to support for your project(s) or organization?
XXX

> #### Why is it currently difficult to achieve this use case?
XXX

### Details

_This is my first issue in this repository (as well as my first in a "large" open-source project) so please forgive me if I missed anything regarding guidelines or style_

The [code](https://github.com/mlflow/mlflow/blob/64df72bff66a2deb5d7b82bc083ec42fd7e7e5a2/mlflow/lightgbm/__init__.py#L496) responsible for the current behaviour:
```python
def _patch_metric_names(metric_dict):
    # lightgbm provides some metrics with "@", e.g. "ndcg@3" that are not valid MLflow metric names
    patched_metrics = {
        metric_name.replace("@", "_at_"): value for metric_name, value in metric_dict.items()
    }
    changed_keys = set(patched_metrics.keys()) - set(metric_dict.keys())
    if changed_keys:
        _logger.info(
            "Identified one or more metrics with names containing the invalid character `@`."
            " These metric names have been sanitized by replacing `@` with `_at_`, as follows: %s",
            ", ".join(changed_keys),
        )

    return patched_metrics
```

I'll have to get more familiar with the code to propose a suitable solution, but my initial suggestion would be to track whether the same log for the same metric has been produced previously in the current run or not.
If this was the case already, we don't log the same message again and again.

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by WeichenXu123** [comment]
I suggest to log it in a more intact way, i.e., we can log one info log for each autologging run, printing all patched metric names in the info log.
